saged/datasets.py

```diff
@@ -731,16 +943,59 @@ def from_paths(class_object,
         new_dataset = class_object(all_expression, sample_to_label, sample_to_study)
         return new_dataset
 
-    def __len__(self) -> int:
+    @classmethod
+    def from_list(class_object,
+                  dataset_list: Sequence["RefineBioLabeledDataset"],
+                  ) -> "RefineBioLabeledDataset":
         """
-        Return the length of the dataset for use in determining
-        the size of an epoch
+        Create a dataset by combining a list of other datasets
+
+        Arguments
+        ---------
+        dataset_list: The list of datasets to combine
 
         Returns
         -------
-        length: The number of samples currently available in the dataset
+        combined_dataset: The dataset created from the other datasets
+
+        Raises
+        ------
+        ValueError: If no datasets are in the list
         """
-        return len(self.current_expression.columns)
+        if len(dataset_list) == 0:
+            raise ValueError('The provided list of datasets is empty')
+
+        elif len(dataset_list) == 1:
+            return dataset_list[0]
+
+        else:
+            combined_dataframe = dataset_list[0].current_expression
+            combined_labels = dataset_list[0].sample_to_label
+            combined_studies = dataset_list[0].sample_to_study
+
+            for dataset in dataset_list[1:]:
+                current_dataframe = dataset.current_expression
+                current_labels = dataset.sample_to_label
+                current_studies = dataset.sample_to_study
+
+                # Ensure no sample gets duplicated
+                # https://stackoverflow.com/questions/19125091/pandas-merge-how-to-avoid-duplicating-columns
+                cols_to_use = current_dataframe.columns.difference(combined_dataframe.columns)
+
+                # Combine dataframes
+                combined_dataframe = pd.merge(combined_dataframe,
+                                              current_dataframe[cols_to_use],
+                                              left_index=True,
+                                              right_index=True,
+                                              how='outer')
+                # Combine labels (double start expands a dictionary into key-value pairs)
```

Comment:
```suggestion
                # Combine labels (double star expands a dictionary into key-value pairs)
```
Also, do you want this comment in the function in `RefineBioUnlabeledDataset` too?

Reply:
Went ahead and added it to `RefineBioUnlabeledDataset`